h/views/activity.py

```diff
@@ -42,9 +42,81 @@ def search(request):
     # Fetch results
     result = query.execute(request, q, page_size=page_size)
 
+    for user in result.aggregations.get('users', []):
+        user['username'] = util.user.split_user(user['user'])['username']
+        del user['user']
+
     return {
         'total': result.total,
         'aggregations': result.aggregations,
         'timeframes': result.timeframes,
         'page': paginate(request, result.total, page_size=page_size),
     }
+
+
+@view_config(route_name='activity.group_search',
+             request_method='GET',
+             renderer='h:templates/activity/search.html.jinja2')
+def group_search(request):
+    if not request.feature('search_page'):
+        raise httpexceptions.HTTPNotFound()
+
+    result = search(request)
+
+    pubid = request.matchdict['pubid']
+
+    try:
+        group = request.db.query(models.Group).filter_by(pubid=pubid).one()
+    except exc.NoResultFound:
+        return result
+
+    if request.authenticated_user not in group.members:
+        return result
+
+    result['group'] = {
+        'created': group.created.strftime('%B, %Y'),
+        'description': group.description,
+        'name': group.name,
+        'pubid': group.pubid,
+    }
+
+    result['show_group_leave_button'] = (
+        group.creator != request.authenticated_user)
```

Comment:
Group creators can leave their own group in the client, so we shouldn't stop them from doing so here. If we decide we want to change the policy around leaving a group, we should do so in a way that makes it consistent whether leaving from the client or the service.